data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/LogstashPluginMapper.java

```diff
@@ -1,16 +1,57 @@
 package org.opensearch.dataprepper.logstash.mapping;
 
 import com.amazon.dataprepper.model.configuration.PluginModel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.opensearch.dataprepper.logstash.exception.LogstashMappingException;
 import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
 
-public interface LogstashPluginMapper {
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
 
-    /**
-     * Takes a Logstash plugin with its associated attributes and maps it to corresponding
-     * Data Prepper plugin and attributes based on Mapping files
-     *
-     * @param logstashPlugin A Logstash plugin with its attributes
-     * @return A Data Prepper plugin with its attributes
-     */
-    PluginModel mapPlugin(LogstashPlugin logstashPlugin);
+/**
+ * Converts Logstash plugin model to Data Prepper plugin model using mapping file
+ *
+ * @since 1.2
+ */
+public class LogstashPluginMapper {
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    private final AttributesMapperProvider attributesMapperProvider;
+
+    public LogstashPluginMapper() {
+        this(new AttributesMapperProvider());
+    }
+
+    LogstashPluginMapper(final AttributesMapperProvider attributesMapperProvider) {
```

Comment:
Why do we have this additional constructor?

Reply:
If it is for unit testing, it would be good to add a Javadoc noting that.